dev/dev-env-setup.sh

```diff
@@ -0,0 +1,232 @@
+
+-q, -quiet        --quiet       Whether to have pip install in quiet mode (Default: false)
+
+EOF
+}
+
+export verbose=0
+export quiet=0
+while getopts "d:fvqh" opt
+do
+  case "$opt" in
+    d) directory="$OPTARG" ;;
+    f) full_install=1 ;;
+    v) verbose=1 ;;
+    q) quiet=1 ;;
+    h) showHelp; exit ;;
+    *) showHelp; exit ;;
+  esac
+done
+
+if [[ $verbose == 1 ]]; then
+  set -exv
+fi
+
+# Acquire the OS for this environment
+case "$(uname -s)" in
+  Darwin*)                       machine=mac;;
+  Linux*)                        machine=linux;;
+  *)                             machine=unknown;;
+esac
+
+function quietpip(){
+  echo $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' )
+}
+
+# Check if pyenv is installed and offer to install it if not present
+pyenv_exist=$(command -v pyenv)
+
+if [ -z "$pyenv_exist" ]; then
+  if [ -z "$GITHUB_ACTIONS" ]; then
+    read -p "pyenv is required to be installed to manage python versions. Would you like to install it? $(tput bold)(y/n)$(tput sgr0): " -n 1 -r
+    echo
+  fi
+  if [[ $REPLY =~ ^[Yy]$ || -n "$GITHUB_ACTIONS" ]]; then
+    if [[ "$machine" == mac ]]; then
+      # Check if brew is installed and install it if it isn't present
+      # Note: if xcode isn't installed, this will fail.
+      if [ -z "$(command -v brew)" ]; then
+        echo "Brew is required to install pyenv on MacOS. Installing in your home directory."
+        bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+      fi
+      echo "Updating brew and installing pyenv..."
+      echo "Note: this will probably take a considerable amount of time."
+      brew update
+      brew install pyenv
+      brew install openssl readline sqlite3 xz zlib
+    elif [[ "$machine" == linux ]]; then
+      sudo apt-get update -y
+      sudo apt-get install -y make build-essential libssl-dev zlib1g-dev \
+      libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
+      libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+      # Install pyenv from source
+      git clone --depth 1 https://github.com/pyenv/pyenv.git "$HOME/.pyenv"
+      PYENV_ROOT="$HOME/.pyenv"
+      PYENV_BIN="$PYENV_ROOT/bin"
+      if [ -n "$GITHUB_ACTIONS" ]; then
+        echo "$PYENV_BIN" >> "$GITHUB_PATH"
+        echo "PYENV_ROOT=$PYENV_ROOT" >> "$GITHUB_ENV"
+      fi
+    else
+      echo "Unknown operating system environment: $machine exiting."
+      exit 1
+    fi
+  else
+    PYENV_README=https://github.com/pyenv/pyenv/blob/master/README.md
+    echo "pyenv is required to use this environment setup script. Please install by following instructions here: $PYENV_README"
+    exit 1
+  fi
+fi
+
+MLFLOW_HOME=$(pwd)
+rd="$MLFLOW_HOME/requirements"
+
+# Get the minimum supported version from MLflow to ensure any feature development adheres to legacy Python versions
+min_py_version=$(grep "python_requires=" "$MLFLOW_HOME/setup.py" | grep -E -o "([0-9]{1,}\.)+[0-9]{1,}")
```

Comment:
```suggestion
min_py_version=$(python setup.py -q min_python_version)
```

Let's use the `min_python_version` command.

Reply:
updated!